src/Persistent/Api.hs

```diff
@@ -353,18 +329,18 @@ instance FromProto Delegation where
     fromProto (ProtoDelegation ctx f t) m = Delegation m ctx f t
 
 -- | So far `mkMetaInfo` is only used by `nextMetaInfo`.
-mkMetaInfo :: User -> Timestamp -> AUID a -> MetaInfo a
-mkMetaInfo user now oid = MetaInfo
+mkMetaInfo :: User -> AUID User -> Timestamp -> AUID a -> MetaInfo a
+mkMetaInfo cUser uid now oid = MetaInfo
```

Comment:
Why do you pass `uid` as a second argument and have an explicit contract that `cUser ^. _id == uid` must always hold?  I would like to change that back if it's ok.